src/VecSim/spaces/L2/L2_SVE_FP64.h

```diff
@@ -0,0 +1,81 @@
+/*
+ *Copyright Redis Ltd. 2021 - present
+ *Licensed under your choice of the Redis Source Available License 2.0 (RSALv2) or
+ *the Server Side Public License v1 (SSPLv1).
+ */
+
+#include "VecSim/spaces/space_includes.h"
+#include <arm_sve.h>
+
+static void L2SquareStep(double *&pVect1, double *&pVect2, size_t &offset, svfloat64_t &sum) {
+    // Load vectors
+    svfloat64_t v1 = svld1_f64(svptrue_b64(), pVect1 + offset);
+    svfloat64_t v2 = svld1_f64(svptrue_b64(), pVect2 + offset);
+
+    // Calculate difference between vectors
+    svfloat64_t diff = svsub_f64_x(svptrue_b64(), v1, v2);
+
+    // Square the difference and accumulate: sum += diff * diff
+    sum = svmla_f64_x(svptrue_b64(), sum, diff, diff);
+
+    // Advance pointers by the vector length
+    offset += svcntd();
+}
+
+template <bool partial_chunk, unsigned char additional_steps>
+double FP64_L2SqrSIMD_SVE(const void *pVect1v, const void *pVect2v, size_t dimension) {
+    double *pVect1 = (double *)pVect1v;
+    double *pVect2 = (double *)pVect2v;
+    size_t offset = 0;
+
+    // Get the number of 64-bit elements per vector at runtime
+    uint64_t vl = svcntd();
+
+    // Multiple accumulators to increase instruction-level parallelism
+    svfloat64_t sum0 = svdup_f64(0.0);
+    svfloat64_t sum1 = svdup_f64(0.0);
+    svfloat64_t sum2 = svdup_f64(0.0);
+    svfloat64_t sum3 = svdup_f64(0.0);
+
+    // Process vectors in chunks, with unrolling for better pipelining
+    auto chunk_size = 4 * vl;
+    size_t number_of_chunks = dimension / chunk_size;
+    for (size_t i = 0; i < number_of_chunks; ++i) {
+        // Process 4 chunks with separate accumulators
+        L2SquareStep(pVect1, pVect2, offset, sum0);
+        L2SquareStep(pVect1, pVect2, offset, sum1);
+        L2SquareStep(pVect1, pVect2, offset, sum2);
+        L2SquareStep(pVect1, pVect2, offset, sum3);
+    }
+
+    if constexpr (additional_steps >= 1) {
+        L2SquareStep(pVect1, pVect2, offset, sum0);
+    }
+    if constexpr (additional_steps >= 2) {
+        L2SquareStep(pVect1, pVect2, offset, sum1);
+    }
+    if constexpr (additional_steps >= 3) {
+        L2SquareStep(pVect1, pVect2, offset, sum2);
+    }
+
+    if constexpr (partial_chunk) {
+        svbool_t pg = svwhilelt_b64(offset, dimension);
+
+        // Load vectors with predication
+        svfloat64_t v1 = svld1_f64(pg, pVect1 + offset);
+        svfloat64_t v2 = svld1_f64(pg, pVect2 + offset);
+
+        // Calculate difference with predication (corrected)
+        svfloat64_t diff = svsub_f64_x(pg, v1, v2);
+
+        // Square the difference and accumulate with predication
+        sum0 = svmla_f64_m(pg, sum0, diff, diff);
```

Comment:
`sum0` was used in a previous step (and in the next)
```suggestion
        sum4 = svmla_f64_m(pg, sum4, diff, diff);
```